src/greatwall/gui.py

```diff
@@ -1023,7 +1023,13 @@ def config_selecting_derivation_widgets(self):
         )
 
         if self.tacit_knowledge_combobox.currentText() == constants.FRACTAL:
-            user_options = self.greatwall.get_fractal_query()
+
+            user_options = self.greatwall.history_fractal_load()
+
+            if user_options is None:
+                user_options = self.greatwall.get_fractal_query()
+                self.greatwall.history_fractal_save(user_options)
```

Comment:
This is a discussion purely about style. The code is readable and works as it is, there is no need for change here.

Reply:
> This is a discussion purely about style.

We have a high standards to let styling may be obstruction to merge code. If we don't agree on the style through a whole project we will end up with code that can't be maintained. In this regard, we have adopted some styles that we need to stick with them, of course we are opening to any suggestion.

However, I'm not taking about code styling, which is also important, I'm taking about API design of the `greatwall` module.

Specifically, I'm talking about hide the methods that are related to internal implementation of the module. For example, the client (someone who uses `greatwall` module) maybe expect he need to call `history_fractal_save(fractal)` to handle the case of saving generated fractals, but this is the responsibility of the `greatwall` module not the client, so we use Information Hiding here to does not expose this method as the interface of the module. Also, the user maybe using it in a wrong way by saving fractals that are not securely generated or has been leaked at some point of his generation to the adversary.

My suggestion is about simplifying the API interface of the `greatwall` module. Primary, what I'm taking about is relating to Information Hiding idea in the the OOP. The `greatwall` module is the core part of our project, if we don't follow the best practice in designing and implementation this may lead to catastrophic issues in the future releases and of course we don't need to maintain compatibility releases because we didn't think a little bit before introduce a new interface to the module. The `gui` module (our GUI desktop implementation using qt and we also plan to implement Android and iOS versions) act as a client for the `greatwall` module, so, we need to implement a clean and clear interface to `greatwall` module, so, we can then use it to design cross-platform GUI applications.

Also, any professional API designing process start by first writing the specifications of the members of this API, so, I encourage you to start by this first. This will minimize and facilitate the process of the reviewing and fixing the bugs. In addition, this will help you to don't deviate from the methods specification and produce a consistent API. Also, this will help us discuss and choose the best API before we go to implementation.

> The code is readable and works as it is, there is no need for change here.

"readable and works" is not enough and strong predicates to jump to "there is no need for change here" conclusion. Instead, address why we don't need to implement what I'm suggest.